### PR TITLE
Media3 local checkout

### PR DESCRIPTION
**Pull request by yschimke**
#### WHAT

See https://github.com/androidx/media#locally

#### WHY

Use a local checkout when testing with modifications or fixes from main.

#### HOW

Uncomment flags in settings.gradle to import media3.

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -45,3 +45,7 @@ include ':network-awareness'
 enableFeaturePreview("VERSION_CATALOGS")
 // https://docs.gradle.org/7.4/userguide/declaring_dependencies.html#sec:type-safe-project-accessors
 enableFeaturePreview("TYPESAFE_PROJECT_ACCESSORS")
+
+// Uncomment to use a local checkout
```
Comment (luizgrp) [inline settings.gradle:49]: nit: could have use a flag from gradle.properties to decide to apply this or not. Flag could also be used for function projectOrDependency

**Review by luizgrp — APPROVED** [review]

File: settings.gradle (hunk 1 above)
Comment (yschimke) [reply settings.gradle:49]: Nice idea. I'm not sure how to do the latter (set a static field?), but first makes sense.

Comment (luizgrp) [reply settings.gradle:49]: you can refer to it directly in the code:
```groovy
def projectOrDependency(module, dependency = null) {
    if (FLAG) {
        return findProject(module)
    } else {
        return dependency
    }
}
```

if you have defined `FLAG` in `gradle.properties`

Comment (yschimke) [reply settings.gradle]: gradle.properties is checked in though, so easy to accidentally commit the change.  What about from local.properties?

Comment (luizgrp) [reply settings.gradle]: local.properties is specific to android studio as far as I know and is cumbersome to load as you probably noticed, seeing your last commit. Trade-offs, maybe leave a comment on top of the flag to not push to remote in a hope it's spotted by code reviewers when changed? other options [here](https://docs.gradle.org/current/userguide/build_environment.html), like environment variables. Or back to original code? happy with whatever you decide

Comment (yschimke) [reply settings.gradle]: I've used local.properties for similar things, so I'll keep that change for now. It avoids editing files under source control.

**Pull request merged**